server/routes/overview/prisonerSupportNeedsRequestHandler.ts

```diff
@@ -0,0 +1,31 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import { CuriousService } from '../../services'
+
+/**
+ * Class exposing middleware functions related to ensuring we have a prisoner's support needs.
+ */
+export default class PrisonerSupportNeedsRequestHandler {
+  constructor(private readonly curiousService: CuriousService) {}
+
+  /**
+   *  Middleware function to look up the prisoner's Support Needs, and store them in the session
+   */
+  getPrisonerSupportNeeds: RequestHandler = async (req: Request, res: Response, next: NextFunction) => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary } = req.session
+
+    try {
+      if (!req.session.supportNeeds) {
+        req.session.supportNeeds = await this.curiousService.getPrisonerSupportNeeds(
+          prisonNumber,
+          prisonerSummary.establishmentId,
+          req.user.username,
+        )
+      }
+      next()
+    } catch (error) {
+      next(createError(404, 'Prisoner Support Needs not found')) // TODO - what should happen here?
```

Comment:
Agreed - we need to work out what to do in the case of a 404 vs. other error status codes
Perhaps leave it for now in this PR, and we do it afterwards (next PR), otherwise this PR could become pretty big?